src/Microsoft.DncEng.Configuration.Extensions/ServiceConfigurationExtensions.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+#if NETCOREAPP2_1
+using IHostEnvironment = Microsoft.Extensions.Hosting.IHostingEnvironment;
+#else
+using IHostEnvironment = Microsoft.Extensions.Hosting.IHostEnvironment;
+#endif
+
+namespace Microsoft.DncEng.Configuration.Extensions
+{
+    public static class ServiceConfigurationExtensions
+    {
+        public static IServiceCollection AddDefaultJsonConfiguration(this IServiceCollection services)
+        {
+            return services.AddSingleton(provider =>
+            {
+                IHostEnvironment env = provider.GetRequiredService<IHostEnvironment>();
```

Comment:
Doesn't this have the same standard/core problem?  If we are building for standard, but running for core, we are going to look for the wrong type.

Reply:
No, we are building for netcoreapp in all cases. And I have aliased the IHostingEnvironment to IHostEnvironment for 2.1 so we can talk about IHostEnvironment in all cases. In 3.1 they deprecated IHostingEnvironment in favor of a new IHostEnvironment type.